### PR TITLE
Fix: User can login to other site with Social Auth

### PR DESCRIPTION
**Pull request by burhandodhy**
**Description**
This PR fixes the issue where user can login to other site using the social authentication

**Jira ticket**
https://edlyio.atlassian.net/browse/EDLY-2269

image

### CODE REVIEW

**Review by zubair-arbi — APPROVED** [review]
Changes looks good 👍

**Review by iam-mhaseeb — APPROVED** [review]

**Review by asadmanzoor93 — APPROVED** [review]

File: openedx/features/edly/validators.py [hunk 1]
```diff
@@ -42,3 +46,25 @@ def is_edly_user_allowed_to_login(request, possibly_authenticated_user):
...
+
+    Arguments:
+        request (object): HTTP request object
+        user: User object trying to authenticate
+
+    Returns:
+        bool: Returns True if User can login to site otherwise False.
+    """
+
+    if not is_edly_user_allowed_to_login(request, user):
```
Comment (ziafazal) [inline openedx/features/edly/validators.py:63]: @burhandodhy can we remove this logic from here and from `login_user` method and move it into `set_logged_in_edly_cookies` method

Comment (burhandodhy) [reply openedx/features/edly/validators.py:63]: @ziafazal  I did move code from `login_user` to `set_logged_in_edly_cookies` but it create an issue. As `set_logged_in_edly_cookies` runs after user is authenticated successfully https://github.com/edly-io/edx-platform/blob/37a3b5167bbfb472018dd118cc531dff5d4e7ba5/openedx/core/djangoapps/user_authn/views/login.py#L370 

So In a case where does user doesn't have specific site access, LMS show the error message as expected but when you click the login button second, `EdlyOrganizationAccessMiddleware` middleware refresh the page as it found that user is already loggedIn.

**Review by ziafazal — APPROVED** [review]
👍

**Pull request merged**